README.md

```diff
@@ -216,18 +297,209 @@ store.firestore
       // Add one person to the city population
       const newPopulation = doc.data().population + 1;
       t.update(cityRef, { population: newPopulation });
-    });
-  })
-  .then(result => {
+    })
+  .then((result) => {
     // TRANSACTION_SUCCESS action dispatched
     console.log('Transaction success!');
   })
-  .catch(err => {
+  .catch((err) => {
     // TRANSACTION_FAILURE action dispatched
     console.log('Transaction failure:', err);
   });
 ```
 
+---
+
+#### Mutations
+
+Mutation operations are immediately reflected in Redux via
+the cache reducer (firestore.cache['some-query'].docs). All
+queries using the collection type will also be reprocessed
+synchronously to opimisticly reflect the update.
+
+Once Firestore has reconciled an operation all opimistic data
+will be removed from the synchronous, in-memory store.
+
+Be advised that in order for transactions to operate
+synchronously the write functions will be processed twice.
+First synchronously using only the in-memory cache and
+then later using Firestore's lazy loading for more accurate data.
+
+```js
+// @required
+const requiredFields = {
+  collection: 'full/path/to/the/collection',
+  doc: 'firestore-document-id',
+};
+
+// changes requested to your data
+const firestoreDocument = {
+  someString: 'string',
+  someNumber: 1,
+  otherNumber: ['::increment', 5],
```

Comment:
Is there a reason we can't have this leverage the existing Firestore API through [`firestore.FieldValue.increment`](https://firebase.google.com/docs/reference/js/firebase.firestore.FieldValue#static-increment) instead of a string that becomes a different part of this API? Same with arrayUnion and arrayRemove

If we really can't and the string is needed we should add a clear explanation as to why - the documentation part doesn't have to block the release necessarily, but would actually like to answer that for myself 😆

Reply:
That’s a fair critique. 

FieldValue has an internal property to store values. Right now it’s possible, but ugly, to dig in and get this property. That property is different for each field value. But if they make it private then it would be impossible. If they change the internals of their API it breaks this library. Also passing class instances that are not serializable in a message is frowned upon in Redux’s rule to have action messages serializable. 

I can update the docs to explain fully.